packages/sdk-middleware-http/src/http.js

```diff
@@ -112,9 +112,11 @@ export default function createHttpMiddleware({
             res.json().then((result: Object) => {
               const parsedResponse: Object = {
                 ...response,
-                body: result,
                 statusCode: res.status,
               }
+
+              if (fetchOptions.method !== 'HEAD') parsedResponse.body = result
```

Comment:
As mentioned before, you need to check for the method **before** parsing the body, as `HEAD` responses do not send a body.

I'm not sure how I can make it more clear. What is confusing you? Can I help you somehow?